cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala

```diff
@@ -0,0 +1,396 @@
+    get.releaseConnection()
+    val fields = IOUtils.toString(resp.entity.get.content, "UTF-8").parseJson.asJsObject.fields
+    var status = "none"
+    if (fields.keySet.contains("modelInfo")) {
+      status = fields("modelInfo").asInstanceOf[JsObject].fields
+        .get("status").map(_.convertTo[String]).get.toLowerCase()
+    } else if (fields.keySet.contains("summary")) {
+      status = fields("summary").asInstanceOf[JsObject]
+        .fields.get("status").map(_.convertTo[String]).get.toLowerCase()
+    }
+    status match {
+      case "ready" | "failed" => Some(resp)
+      case "created" | "running" => None
+      case s => throw new RuntimeException(s"Received unknown status code: $s")
+    }
+  }
+
+  protected def handlingFunc(client: CloseableHttpClient,
+                             request: HTTPRequestData): HTTPResponseData = {
+    val response = HandlingUtils.advanced(getBackoffs: _*)(client, request)
+    if (response.statusLine.statusCode == 201) {
+      val location = new URI(response.headers.filter(_.name == "Location").head.value)
+      val maxTries = getMaxPollingRetries
+      val key = request.headers.find(_.name == "Ocp-Apim-Subscription-Key").map(_.value)
+      val it = (0 to maxTries).toIterator.flatMap { _ =>
+        queryForResult(key, client, location).orElse({
+          blocking {
+            Thread.sleep(getPollingDelay.toLong)
+          }
+          None
+        })
+      }
+      if (it.hasNext) {
+        it.next()
+      } else {
+        throw new TimeoutException(
+          s"Querying for results did not complete within $maxTries tries")
+      }
+    } else {
+      response
+    }
+  }
+}
+
+object MultivariateAnomalyModel extends ComplexParamsReadable[MultivariateAnomalyModel] with Serializable
+
+class MultivariateAnomalyModel(override val uid: String) extends Estimator[DetectMultivariateAnomaly]
+  with MADBase {
+  logClass()
+
+  def this() = this(Identifiable.randomUID("MultivariateAnomalyModel"))
+
+  def urlPath: String = "anomalydetector/v1.1-preview/multivariate/models"
+
+  val slidingWindow = new ServiceParam[Int](this, "slidingWindow", "An optional field, indicates" +
+    " how many history points will be used to determine the anomaly score of one subsequent point.", {
+    case Left(x) => (x >= 28) && (x <= 2880)
+    case Right(_) => true
+  }, isRequired = true)
+
+  def setSlidingWindow(v: Int): this.type = setScalarParam(slidingWindow, v)
+
+  def setSlidingWindowCol(v: String): this.type = setVectorParam(slidingWindow, v)
+
+  def getSlidingWindow: Int = getScalarParam(slidingWindow)
+
+  def getSlidingWindowCol: String = getVectorParam(slidingWindow)
+
+  val alignMode = new ServiceParam[String](this, "alignMode", "An optional field, indicates how " +
+    "we align different variables into the same time-range which is required by the model.{Inner, Outer}", {
+    case Left(s) => Set("Inner", "Outer")(s)
```

Comment:
nit: Might want to make this case insensitive

Reply:
Done.